### DISCUSSION TITLE
What are your priorities in terms of documentation?

### DISCUSSION THREAD

**Discussion by mdouze** [facebookresearch/faiss#3685] [General]
(re-posting here for visibility)

Dear Faiss users,
In 2021 we are planning to work on improvements of the Faiss documentation. Please comment here to say what you think is most important for your usage. Different axes we are thinking of:

1. better tutorial
2. better information on how to compile or install Faiss (please be specific)
3. Python docstrings for interactive use in notebooks
4. doxygen documentation: make more accessible, improve the doxygen annotations
5. definition of what is the public / private API, where public is guaranteed not to change
6. more structured presentation of the index types
7. more benchmarks
8. detail existing topics in the wiki or add other topics (eg. Faiss as a database)
9. explicit roadmap
10. help for porting to other platforms
11. C API documentation
12. an interactive forum to get help from (like https://discuss.pytorch.org/)
13. anything else?

We don't make any promise, but we will try to take feedback into account!

**Comment by bozheng-hit** [comment]
It seems that the documentation page is down https://rawgit.com/facebookresearch/faiss/master/docs/html/annotated.html.

**Comment by wickedfoo** [comment]
@bozheng-hit  I don't know what that rawgit.com link is (did someone else create it?), but our existing documentation has always been here:

https://github.com/facebookresearch/faiss/wiki

**Comment by beauby** [comment]
@bozheng-hit The new docs website is available at https://facebookresearch.github.io/faiss

**Comment by bozheng-hit** [comment]
> @bozheng-hit The new docs website is available at https://facebookresearch.github.io/faiss

Thanks a lot! Will the python APIs be available as well?

**Comment by beauby** [comment]
@bozheng-hit Yes, we are working on it

**Comment by ckald** [comment]
I find the installation of faiss without Anaconda very difficult, I'd love to see documentation for that

**Comment by msdejong** [comment]
The two difficulties I have encountered when using FAISS are:
1. When compiling from source, I ran into a number of gremlins (SWIG version, library linking) and when it finally compiled, I still encountered import errors that did not really give a hint as to what was going on. Ultimately gave up. Not an expert, but have compiled a good number of other challenging packages from source, so I think it's fair to say it's not easy. It would help to watch 4 or 5 semi-clueless people try to compile from source, and write down a solution for every problem they encounter on the way. 

2. There is good information available on what index to choose for what use case, but it was difficult to find what parameters different indices have and what they mean. All the information is available in the wiki somewhere, but a generously annotated API would be very helpful here. I know there is https://github.com/facebookresearch/faiss/wiki/Faiss-indexes, which is useful but lacking some explanation/detail

**Comment by KunmingS** [comment]
> @bozheng-hit Yes, we are working on it

It's nice to know you're working with python API! About what time the python API can be available?

**Comment by mdouze** [comment]
There is an open branch here: 

https://github.com/facebookresearch/faiss/tree/docstring-comments

when all docstrings will be filled in, we'll make a PR of it.
